internal/pkg/scorecard/test_definitions.go

```diff
@@ -105,3 +106,40 @@ func NewTestSuite(name, description string) *TestSuite {
 		Weights: make(map[string]float64),
 	}
 }
+
+// MergeSuites takes an array of TestSuites and combines all suites with the same name
+func MergeSuites(suites []*TestSuite) ([]*TestSuite, error) {
```

Comment:
Suggest using `[]TestSuite` for both input and returned values, and also moving `[]*TestResult` to `[]TestResult` in the `TestSuite` struct and in the various places in this function.